src/seedu/addressbook/data/person/Address.java

```diff
@@ -2,43 +2,63 @@
 
 import seedu.addressbook.data.exception.IllegalValueException;
 
+import static seedu.addressbook.data.person.Block.isValidBlock;
+import static seedu.addressbook.data.person.PostalCode.isValidPostalCode;
+import static seedu.addressbook.data.person.Street.isValidStreet;
+import static seedu.addressbook.data.person.Unit.isValidUnit;
+
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "666, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in 'BLOCK, STREET, UNIT, POSTAL_CODE' format without quotes. Please see 'help' for more information.";
 
     public final String value;
     private boolean isPrivate;
 
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
```

Comment:
👍 For making these private. However, they should be grouped together with other instance variables. (e.g. the ones above).